Eleino/feature/557 teacher main page implementation

eleino opened the pull request:
## 📄 **Pull Request Overview**

Closes #557 

## 🔧 **Changes Made**

1. Updated the Teacher main page to match figma design
2. Added a new custom Button element
3. Adding and deleting lessons works on front-end side, max of 10 lessons
4. Added translations for Teacher page


---

## ✅ **Checklist Before Submission**

- **Functionality**: I have tested my code, and it works as expected.
- **JSDoc**: I have added or updated JSDoc comments for all relevant code.
- **Debugging**: No `console.log()` or other debugging statements are left.
- **Clean Code**: Removed commented-out or unnecessary code.
- **Tests**: Added new tests or updated existing ones for the changes made.
- **Documentation**: Documentation has been updated (if applicable).

---

## 📝 **Additional Information**

Provide any additional context or information that reviewers may need to know:

- **Screenshots**:
- 
<img width="1555" height="949" alt="kuva" src="https://github.com/user-attachments/assets/34794653-1c78-4c14-a361-e56072ae2997" />
<img width="1339" height="895" alt="kuva" src="https://github.com/user-attachments/assets/513896c5-4d97-46c4-ac3b-6cd84937efaa" />
<img width="1322" height="793" alt="kuva" src="https://github.com/user-attachments/assets/875c49bd-750b-413f-aa88-1d02d9a3a2a4" />

<img width="1188" height="661" alt="kuva" src="https://github.com/user-attachments/assets/d4ab9ead-05f7-4a50-9587-9dc43665ff0f" />

<img width="1169" height="660" alt="kuva" src="https://github.com/user-attachments/assets/2ff1ead6-aea8-48a2-a2b1-3d1f382b9dc7" />

- **Dependencies**: [Mention any new dependencies or breaking changes]
- **Known Issues**: 
- No backend communication, currently using mock data with the information available via figma design, the final form of the data and even the page itself is still unknown.
- Also, there are some old unfinished teachers' page modules under app/[lng]/[helper]/teachers I did not touch (gaming, instructions, preparation, results).

eleino commented:
Also: there is no link to the page, but it's at /fi/teachers (or /en/teachers).

Rutjake reviewed (changes requested):
Good work! The site looks great and functions dynamically. This structure is excellent; it ensures that site enhancement can continue easily when implementing the backend in the future.

Requested Revisions:

1. Font Variables
I addressed the font variable usage as an example in a separate comment, but please check it in all edited files. Remember to use the project's global SCSS variables for consistency.

2. Layout Enhancement (Max Width)
I have a minor enhancement suggestion for the layout:
Currently, the content sections are wider than the main navigation bar on larger screens (2K resolution and above). It would look much nicer and more cohesive if the main page content were set to the same maximum width as the navigation bar (e.g., 1400px). This will create a cleaner visual alignment.

Skoivumaki commented on frontend-next-migration/src/preparedPages/TeachersPage/ui/LessonList/LessonList.module.scss:
```diff
@@ -0,0 +1,28 @@
+    h2 {
+        font-family:var(--font-family-title);
+        font-weight: 400;
+        font-size:var(--h3-size-desktop);
```
See our Wiki for even better explanation :smile: 
https://github.com/Alt-Org/Altzone-WebPages/wiki/9.-Css-rules

eleino commented:
Thanks! I've updated the TeachersPage to use the same Container component NavBar uses, that should ensure it stays the same size relative to it. I've also changed the css font definitions to use the global definitions.

Skoivumaki commented on frontend-next-migration/src/preparedPages/TeachersPage/ui/LessonList/LessonList.tsx:
```diff
@@ -0,0 +1,83 @@
+                    {t('create-lesson')}
+                    <Image
+                        src={plusIcon}
+                        alt="+"
```
This is not how `alt` attributes are used. The idea with `alt` is to describe what the input will do and not what it looks like in this context.

>1. For all input elements that have a type attribute value of image, check for the presence of an alt attribute.
>2. Check that the value of the alt attribute describes the button's function.

See: https://www.w3.org/WAI/WCAG21/Techniques/html/H36

Now of course the page doesn't expect the user to be blind, but this is good practice to learn especially if you were to work on a government- or a project that legally is required to comply with WCAG. Overall, understanding accessibility is a key element in web development.

eleino replied:
Sorry, fixed! I thought it was handy to show the plus sign even if the icon wasn't working but you're right that's not the idea with alt text.

Skoivumaki commented on frontend-next-migration/src/shared/ui/AppLink/AppLink.tsx:
```diff
@@ -18,7 +18,7 @@ interface AppLinkProps {
     role?: string;
     ariaLabel?: string;
-    onClick?: () => void;
+    onClick?: React.MouseEventHandler<HTMLAnchorElement | HTMLButtonElement>;
```
What was this change for and why was it required? A commit message could have explained this.
Reason why I'm asking is this is a component which is used a lot, what if the change broke something? did the developer test the change thoroughly? etc...
If I had to guess, its probably to stop browser from throwing event errors. :sweat_smile:

Skoivumaki commented on frontend-next-migration/src/shared/ui/v2/Button/ui/Button.tsx:
Creating a new component of the button wasn't part of the issue, but its nice to see someone finally did it. (old one was overly complex and quite old)
